ipvm/src/network/client.rs

```diff
@@ -110,43 +113,70 @@ impl Client {
     }
 }
 
+/// Wrapper-type for file request name.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct FileRequest(pub(crate) String);
 
+/// Wrapper-type for file response content/bytes.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct FileResponse(pub(crate) Vec<u8>);
 
 #[derive(Debug)]
+/// [Client] commands.
 pub enum Command {
+    /// Start listening on an address.
     StartListening {
+        /// Address to listen on.
         addr: Multiaddr,
+        /// Channel to send on.
         sender: oneshot::Sender<Result<()>>,
     },
+    /// Dial a peer in the cluster.
     Dial {
+        /// Peer identifier.
         peer_id: PeerId,
+        /// Peer address.
         peer_addr: Multiaddr,
+        /// Channel to send on.
         sender: oneshot::Sender<Result<()>>,
     },
+    /// Start providing content over channel.
     StartProviding {
+        /// File.
         file_name: String,
+        /// Channel to send on.
         sender: oneshot::Sender<Result<()>>,
     },
+    /// Lookup providers for given key (file).
     GetProviders {
+        /// File.
         file_name: String,
+        /// Channel to send on.
         sender: oneshot::Sender<Result<HashSet<PeerId>>>,
     },
+    /// Request file from peer.
     RequestFile {
+        /// File.
         file_name: String,
+        /// Peer identifier.
         peer: PeerId,
+        /// Channel to send on.
         sender: oneshot::Sender<Result<Vec<u8>>>,
     },
+    /// Respond with file content.
     RespondFile {
+        /// File content.
         file: Vec<u8>,
+        /// Channel to send on.
         channel: ResponseChannel<FileResponse>,
     },
+    /// Publish message on the topic name.
     PublishMessage {
+        /// Pubsub (i.e. [libp2p::gossipsub]) topic.
         topic: String,
```

Comment:
Super minor thing, and possibly Haskell-brain happening, but would it be helpful to track that this is not just any string with a newtype `Topic`?

Reply:
Yeah, a newtype may work. I won't touch `File` and such yet b/c I think the whole networking layer needs a nice refactor anyhow. It's still a lot of the example code tbh (w/ additions).